### PR TITLE
Octree and minor changes to hit detection logic

### PR DESCRIPTION
**Pull request by rgarmsen2295**
Hey all, already talked ad nauseam about the octree, but here it is.  Ended up needing to fix some issues, so sorry for the delay.

Still got some minor work to do when we introduce multiple levels (rebuilding the tree works in theory but I still need to test it thoroughly).  I know we walked through the code together, but take another look if you get a chance, especially at the additional changes such as returning `nullptr` rather than `NO_OBJECT` when no collision occurs.

Will continue to test on my own as well to make sure everything is in order; I'm sure there are a few "silent" (e.g. non-noticeable but octree ain't working 100% as expected under the hood) still around.  Please let me know if you run into any oddities you think may be octree related and yell if there's a blocker! :D

Resource used for Octrees:
https://www.gamedev.net/resources/_/technical/game-programming/introduction-to-octrees-r3529

### CODE REVIEW

File: src/CookiePhysicsComponent.cpp [hunk 1]
```diff
@@ -41,50 +41,49 @@ void CookiePhysicsComponent::updatePhysics(float deltaTime) {
...
-            n = glm::vec3(1.0, 0.0, 0.0);
-        }
-        if((objBB->min_.y - epsilon <= cookieBB->max_.y && objBB->min_.y + epsilon >= cookieBB->max_.y) ||
-                (objBB->max_.y - epsilon <= cookieBB->min_.y && objBB->max_.y + epsilon >= cookieBB->min_.y)){
-            n = glm::vec3(0.0, 1.0, 0.0);
-        }
-        if((objBB->min_.z - epsilon <= cookieBB->max_.z && objBB->min_.z + epsilon >= cookieBB->max_.z) ||
-                (objBB->max_.z - epsilon <= cookieBB->min_.z && objBB->max_.z + epsilon >= cookieBB->min_.z)){
-            n = glm::vec3(0.0, 0.0, 1.0);
+    if (objHit != nullptr) {
```
Comment (joncatanio) [inline src/CookiePhysicsComponent.cpp:44]: Much better.

**Review by joncatanio — APPROVED** [review]
Looks good Reed, I know we went over this together so I gave it another quick through and it looks good. I'm glad we ended up changing the checkCollision to return a nullptr instead of enum. We need to be careful merging this and #73 because I think we are going to get some conflicts, shouldn't be too crazy though.

File: src/Shape.cpp [hunk 2]
```diff
@@ -339,7 +339,7 @@ void Shape::findAndSetMinAndMax() {
 	float maxX, maxY, maxZ;
 
 	minX = minY = minZ = FLT_MAX;
-	maxX = maxY = maxZ = FLT_MIN;
+	maxX = maxY = maxZ = -FLT_MAX;
```
Comment (nurgan) [inline src/Shape.cpp:342]: much better than 0

**Review by nurgan — APPROVED** [review]

File: src/Shape.cpp (hunk 2 above)
Comment (rgarmsen2295) [reply src/Shape.cpp:342]: Agreed!

**Pull request merged**